gtsam/hybrid/HybridConditional.h

```diff
@@ -0,0 +1,99 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ *  @file HybridConditional.h
+ *  @date Mar 11, 2022
+ *  @author Fan Jiang
+ */
+
+#pragma once
+
+#include <gtsam/hybrid/HybridFactor.h>
+#include <gtsam/discrete/DiscreteConditional.h>
+#include <gtsam/inference/Conditional.h>
+
+
+#include <boost/make_shared.hpp>
+#include <boost/shared_ptr.hpp>
+#include <string>
+#include <vector>
+
+namespace gtsam {
+
+/**
+ * Hybrid Conditional Density
+ *
+ * As a type-erased variant of:
+ * - DiscreteConditional
+ * - GaussianMixture
+ * - GaussianConditional
+ */
```

Comment:
HybridConditional is a type-erased wrapper for all:
- DiscreteConditional
- GaussianMixture
- GaussianConditional

Gist is that by unifying all these types the only work need in HybridBayesTree is boilerplate (same to linear and discrete), the hard work (type-based dispatch) is offloaded to this class.

Reply:
This eliminates the necessity of `AbstractConditional`, which is good because now we don't need to touch other modules, especially `inference`, at all.